vcd/resource_vcd_vm_internal_disk_test.go

```diff
@@ -0,0 +1,468 @@
+// +build vapp vm ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccVcdVmInternalDisk(t *testing.T) {
+	// Thus it won't run in the short test
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	// In general VM internal disks works with Org users, but since we need to create VDC with disabled fast provisioning value, we have to be sys admins
+	if !usingSysAdmin() {
+		t.Skip("VM internal disks tests requires system admin privileges")
+		return
+	}
+
+	/*	adminVdc, err := getAdminVdc()
+		if err != nil {
+			t.Skip(fmt.Sprintf("error retrieving if VDC is thing provisioned %s", err))
+			return
+		}
+
+		vapp, err := getAvailableVapp()
+		if err != nil {
+			t.Skip("No suitable vApp found for this test")
+			return
+		}
+		var vm *govcd.VM
+
+		if vapp.VApp.Children != nil && len(vapp.VApp.Children.VM) > 0 {
+			vm, err = vapp.GetVMById(vapp.VApp.Children.VM[0].ID, false)
+			if err != nil {
+				t.Skip(fmt.Sprintf("error retrieving VM %s", vapp.VApp.Children.VM[0].Name))
+				return
+			}
+		}
+		if vm == nil {
+			t.Skip(fmt.Sprintf("No VM available in vApp %s", vapp.VApp.Name))
+			return
+		}
+
+		// for test to run correctly VM has to be power on
+		task, err := vm.PowerOn()
+		if err != nil {
+			t.Skip(fmt.Sprintf("Error powering up vm %s", err))
+			return
+		}
+		_ = task.WaitTaskCompletion()*/
+	internalDiskSize := 20000
+
+	storageProfile := testConfig.VCD.ProviderVdc.StorageProfile
+	/*	if *adminVdc.UsesFastProvisioning {
+		// to avoid `Cannot use multiple storage profiles in a fast-provisioned VDC` we need to reuse VM storage profile
+		storageProfile = vm.VM.StorageProfile.Name
+	}*/
+
+	diskResourceName := "disk1"
+	diskSize := "13333"
+	biggerDiskSize := "14333"
+	busType := "sata"
+	busNumber := "1"
+	unitNumber := "0"
+	allowReboot := true
+
+	vappName := "TestInternalDiskVapp"
+	vmName := "TestInternalDiskVm"
+	vdcName := "ForInternalDiskTest"
+	var params = StringMap{
+		"Org": testConfig.VCD.Org,
+		//"VDC": testConfig.VCD.Vdc,
```

Comment:
Should be removed?

Reply:
will remove when I sure we finished changing design